client/keys/migrate.go

```diff
@@ -49,7 +50,7 @@ func runMigrateCmd(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	defer legacyKb.Close()
+	defer sdkerrors.CallbackLog(legacyKb.Close)
```

Comment:
Honestly all this error handling of `p.Close()` calls seems redundant - so does `CallbackLog`

Reply:
It's better to check it everywhere then copy-paste and forget to close it in places where it can be critical.

I remember I was reading about some critical bug in a software of one of the big companies (probably Cloudflare) because of that.

Reply:
Look, I agree that many `Close()` calls should be checked rather than unconditionally `defer`d (see _closing files that are open with RW perms_ - seriously please, [Google it](https://www.google.com/search?q=golang+defer+open+files+close)). Yet this does **not solve any problem**. Instead it just adds a useless `if err != nil; println(err)` function call for **the sake of fixing a golangci-lint's pedantic error**. Dulcis in fundo, it **prints errors to /dev/stdout** (which is incidentally **wrong** as per POSIX's recommendations)?

Please revert - or define a simple `CallbackLog`-like [1] function in this package's local scope


[1] How is this function name's supposed to assist the reader by the way, considering that there is neither a `log` (it prints to /dev/stdout) nor a callback involved?

Reply:
Alessio, sorry, I don't see your point. Firstly you say that it's redundant to handle errors from `p.Close` and now you say that I add useless println. 

Logging is better than nothing. At least we will see that something is wrong. I can hack it to assign an error to a return value. 
Changing a print to use `stderr` is an easy change. I will do that.

The meaning of `CallbackLog` is that it creates a callback function which logs an error. It's in `skderrors` package, so the error is already bounded to the package name. Not sure why should I move it to a local package? I'm planning to use it in other places as well. If you have a better name in mind (which doesn't consist of Helper, Util etc..) then let's consider it.

Reply:
> Logging is better than nothing.

Writing Golang-idiomatic code is **better than anything else**. This reduces tiny code duplication for the cost of **losing control** of what's on.

This is Go. Not Java. The client code **must check for errors**. If the client app developers don't like it, they can write their application in Java.

Please revert.

Reply:
@alessio , what are you writing? The error is totally ignored in the original code. There is no way a user can check for an error in `master`.